pilot/pkg/bootstrap/server.go

```diff
@@ -1046,15 +1040,14 @@ func (s *Server) maybeCreateCA(caOpts *caOptions) error {
 			}
 		}
 		// May return nil, if the CA is missing required configs - This is not an error.
-
-		// TODO: Issue #27606 If External CA is configured, use that to sign DNS Certs as well. IstioCA need not be initialized
-		if s.CA, err = s.createIstioCA(corev1, caOpts); err != nil {
-			return fmt.Errorf("failed to create CA: %v", err)
-		}
 		if caOpts.ExternalCAType != "" {
 			if s.RA, err = s.createIstioRA(s.kubeClient, caOpts); err != nil {
 				return fmt.Errorf("failed to create RA: %v", err)
 			}
+		} else {
```

Comment:
don't think this is a good idea. I think if an ExternalCA is configured, we should still use it just to sign workloads certificates and not Istiod's certificate unless the PilotCertProvider is explicitly set to Kubernetes. Note that the External CA's root needs to be distributed to workloads if they are to talk to Istiod that is signed by an external CA

Reply:
Thanks!  changed to If PilotCertProvider=Kubernetes && singerName is configured we will not create CA now.